test/functional/terminal/ex_terminal_spec.lua

```diff
@@ -131,6 +131,14 @@ describe(':terminal (with fake shell)', function()
     end)
   end)
 
+  it(":stopinsert RPC request during insert-mode", function()
+    command(':terminal')
+    feed("i[tui] insert-mode")
+    eq({ blocking=false, mode='t' }, nvim('get_mode'))
+    command('stopinsert')
```

Comment:
This fails correctly before your patch, because all of the `test/functional` tests run from a Lua RPC client to Nvim. So `command()` makes an RPC call.